### ISSUE TITLE
Complex Waveform that has been sliced in (ell,m) can not be interpolated in time

### ISSUE THREAD

**Issue by duetosymmetry** [moble/scri#9]
Because scipy.interpolate functions only act on floats, not complex, WaveformBase.interpolate contains the following code to go through real views:

https://github.com/moble/scri/blob/72aaa1aa8ea4a86b4435f2a6bcef95d02a31ac97/waveform_base.py#L807-L810

However, if the Waveform was built by slicing a WaveformModes in (ell,m), then data_2d will not be C-contiguous. Then np.ndarray.view will raise a ValueError, [as mentioned here](https://www.numpy.org/devdocs/reference/generated/numpy.ndarray.view.html).

**Comment by duetosymmetry** [comment]
What is the penalty for doing:

```
        if self.data.dtype == np.dtype(complex):
            for i in range(W.n_data_sets):
                W.data_2d[:, i] = (splev(W.t, splrep(self.t, self.data_2d[:, i].real, s=0), der=0)
                                   + 1j*splev(W.t, splrep(self.t, self.data_2d[:, i].imag, s=0), der=0))
```

which avoids the error?

**Comment by duetosymmetry** [comment]
I have confirmed that with the version of numpy that I have installed (1.16.3 or later), `.real` and `.imag` make shallow copies, even with sliced arrays. It is rather easy to check. Below is an example session showing this. This is also a minimal working example for the bug above and the fix in #10.
Therefore there is no penalty for using `.real` and `.imag`.

```python
>>> a = np.array([[1., 2., 3.],[7.,8.,9.]]) + np.array([[4., 5., 6.],[10.,11.,12.]])*1.j
>>> print(a)
[[1. +4.j 2. +5.j 3. +6.j]
 [7.+10.j 8.+11.j 9.+12.j]]
>>> d = a[:, 0:2]
>>> print(d)
[[1. +4.j 2. +5.j]
 [7.+10.j 8.+11.j]]
>>> d.base is a
True
>>> d.data.c_contiguous
False
>>> dv = d.view(dtype=float)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-83-eaec6c6460fd> in <module>
----> 1 dv = d.view(dtype=float)

ValueError: To change to a dtype of a different size, the array must be C-contiguous
>>> dr = d.real
>>> di = d.imag
>>> (dr.base is a) and (di.base is a)
True
>>> print(di)
[[ 4.  5.]
 [10. 11.]]
>>> a[0,0] = -a[0,0]
>>> print(a)
[[-1. -4.j  2. +5.j  3. +6.j]
 [ 7.+10.j  8.+11.j  9.+12.j]]
>>> print(di)
[[-4.  5.]
 [10. 11.]]
```

**Comment by moble** [comment]
Yeah, I'm not really sure why I used that strained syntax, but this looks much more sensible.
